src/Contract/Ast/TaggedTokenReferenceInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Contract\Ast;
+
+/**
+ * Represents the AST-Token, its location, and associated tags.
+ */
+interface TaggedTokenReferenceInterface extends TokenReferenceInterface
+{
+    public function hasTag(string $name): bool;
+
+    /**
+     * @return string[]|null
+     */
+    public function getTagLines(string $name): ?array;
```

Comment:
This seems to be only used internally in `layer`. Does it have to be exposed as part of the `Contract`?

Reply:
I converted the trait into a base class, but kept it internal in Core, and kept the interface in Contract.
If we exposed the base class in Contract it's no longer an implementation detail that can easily be changed later.